src/main/java/models/dao/impl/CureDaoFiles.java

```diff
@@ -0,0 +1,257 @@
+package main.java.models.dao.impl;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import main.java.models.Cure;
+import main.java.models.Nurse;
+import main.java.models.Pokemon;
+import main.java.models.PokemonStatus;
+import main.java.models.PokemonType;
+import main.java.models.Trainer;
+import main.java.models.dao.CureDao;
+import main.java.models.dao.DaoFactory;
+import main.java.models.dao.impl.exception.CureNotFoundException;
+import main.java.models.dao.impl.exception.LocalDBIOException;
+import main.java.models.dao.impl.exception.UserNotFoundException;
+
+public class CureDaoFiles implements CureDao {
+
+  private final String path = "./localStorage/cures.ser";
+  private ArrayList<Cure> cures;
+  private ArrayList<Cure> backup;
+
+  public CureDaoFiles() {
+
+    File file = new File(path);
+
+    if (file.exists()) {
+      readFile();
+    } else {
+
+      cures = new ArrayList<>();
+
+      Trainer ash = new Trainer("Ash", "Pallet", "male", "ash", "password", "email");
```

Comment:
Esses códigos aqui no construtor só devem existir caso for uma informação que tem que ser criada na inicialização do projeto, tipo o login do admin. Essas informações que tu colocou aqui são de teste então deveriam estar no main